From d024a3e0e25c2a1bc23f52ca2cb91eef86041c22 Mon Sep 17 00:00:00 2001
From: Deepali Chourasia <deepch23@in.ibm.com>
Date: Tue, 6 Jun 2023 10:32:48 +0000
Subject: [PATCH] fix for gcc 8

---
 src/include/nvtx3/nvtx3.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/include/nvtx3/nvtx3.hpp b/src/include/nvtx3/nvtx3.hpp
index cb0ef68..1da3cd5 100644
--- a/src/include/nvtx3/nvtx3.hpp
+++ b/src/include/nvtx3/nvtx3.hpp
@@ -562,7 +562,8 @@
 /* Temporary helper #defines, removed with #undef at end of header */
 
 #if !defined(NVTX3_USE_CHECKED_OVERLOADS_FOR_GET)
-#if defined(_MSC_VER) && _MSC_VER < 1914
+#if defined(_MSC_VER) && _MSC_VER < 1914 \
+  || defined(__GNUC__) && __GNUC__ == 8 && __GNUC_MINOR__ < 4
 /* Microsoft's compiler prior to VS2017 Update 7 (15.7) uses an older parser
  * that does not work with domain::get's specialization for domain::global,
  * and would require extra conditions to make SFINAE work for the overloaded
-- 
2.34.1

